benchmark_runner/common/remote_ssh/remote_ssh.py

```diff
@@ -0,0 +1,195 @@
+        """
+
+        for file_name in os.listdir(local):
+            if os.path.isfile(os.path.join(local, file_name)):
+                self.get_remote_file(remote, local, file_name)
+            else:
+                self.mkdir(remote, ignore_existing=True)
+                self.get_remote_file(remote, local, file_name)
+
+    def put_remote_file(self, local, remote, file_name):
+        """
+        This method upload file name from local to remote path
+        :param local:
+        :param remote:
+        :param file_name:
+        :return:
+        """
+        self.__p_sftp.put(f'{local}/{file_name}', f'{remote}/{file_name}')
+
+    def get_remote_file(self, remote, local, file_name):
+        """
+        This method download file name from remote to local path
+        :param local:
+        :param remote:
+        :param file_name:
+        :return:
+        """
+        self.__p_sftp.get(f'{remote}/{file_name}', f'{local}/{file_name}')
+
+    def replace_parameter(self, remote_path, file_name, parameter, value, all_line=False):
+        """
+        This method is replace parameter in remote file
+        :param remote_path: path to file
+        :param file_name: file name
+        :param parameter: the parameter
+        :param value: the new value
+        :param all_line: replace all the line that contains the parameter with value
+        :return:
+        """
+        if all_line:
+            res = False
+            for c in ['/', '@', '#', '$', '%', '^', '&', '*', '-', '<', '>', '!', '~', '+', '-']:
+                if c not in parameter:
+                    res = True
+                    self.run_command(f'sed -i \'s{c}{parameter}{c}{value}{c}g\' {remote_path}/{file_name}')
+                    break
+            if not res:
+                raise RunCommandError
+        else:
+            if os.path.splitext(file_name)[1] == '.xml':
+                self.run_command(f'sed -i \'s/{parameter}[^ ]*"/{parameter}{value}/g\' {remote_path}/{file_name}')
+            else:
+                self.run_command(f'sed -i \'s/{parameter}[^ ]*/{parameter}{value}/g\' {remote_path}/{file_name}')
+
+    def exist(self, remote_path):
+        """
+        This method check if file or dir exists on remote
+        :param: remote_path to file or dir:
+        :return: True if exists otherwise False
+        """
+        try:
+            self.__p_sftp.stat(remote_path)
+            return True
+        except IOError:
+            return False
+
+    def rename(self, old_remote_path, new_remote_path):
+        """
+        This method rename remote dir
+        :param old_remote_path:
+        :param new_remote_path:
+        """
+        self.__p_sftp.rename(old_remote_path, new_remote_path)
+
+    def rmdir(self, remote_path):
+        """
+        This method delete entire remote folder recursively
+        """
+        if self.exist(remote_path) and len(remote_path) > 10:
+            self.run_command(f'rm -r {remote_path}')
```

Comment:
It needs to be `rm -rf` or it might get hung up if there are permissions issues.  Furthermore, it should be

```
rm -rf --
```
in case there's a file/directory whose name starts with `-`.  And you need to worry about quoting in case there are any metacharacters/spaces what have you in the filename.  Can't you do it with sftp?

Reply:
:+1: